lib/rules/declaration-block-no-shorthand-property-overrides/index.js

```diff
@@ -32,10 +32,12 @@ const rule = (primary) => {
 
 			eachDecl((decl) => {
 				const prop = decl.prop;
-				const unprefixedProp = vendor.unprefixed(prop);
+				const unprefixedProp = vendor.unprefixed(prop).toLowerCase();
 				const prefix = vendor.prefix(prop).toLowerCase();
 
-				const overrideables = shorthandData[unprefixedProp.toLowerCase()];
+				const overrideables = /** @type {Record<string, Set<string>>} */ (shorthandData)[
+					unprefixedProp
+				];
```

Comment:
[note] Without this type-cast using `@type`, TypeScript compiler complains as below:

```
lib/rules/declaration-block-no-shorthand-property-overrides/index.js:38:27 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ margin: Set<string>; padding: Set<string>; background: Set<string>; font: Set<string>; border: Set<string>; 'border-top': Set<string>; 'border-bottom': Set<string>; ... 27 more ...; mask: Set<...>; }'.
  No index signature with a parameter of type 'string' was found on type '{ margin: Set<string>; padding: Set<string>; background: Set<string>; font: Set<string>; border: Set<string>; 'border-top': Set<string>; 'border-bottom': Set<string>; ... 27 more ...; mask: Set<...>; }'.

38     const overrideables = (shorthandData)[
                             ~~~~~~~~~~~~~~~~
39      unprefixedProp
   ~~~~~~~~~~~~~~~~~~~
40     ];
   ~~~~~
```